### DISCUSSION TITLE
How do I use MuJoCo's visualizer with state data without running the simulation (Python Bindings version 2.3.1)?

### DISCUSSION THREAD

**Discussion by nchen9191** [google-deepmind/mujoco#710] [General]
Hello,

MuJoCo newbie here. I am trying to use MuJoCo's visualizer without needing to run the simulator. I have an XML file that defines the model (cylindrical rod with two springs holding up the ends). Ideally, I want to be able to give it state data and have MuJoCo's visualizer render a frame of that state.

As an exercise, I ran a simulation of this model in MuJoCo and was able to produce a video of the simulation just fine. I then saved the time, qpos, qvel data from MjData. I tried recreate the video by setting MjData's time, qpos, and qvel with the saved data at the timesteps I want to render, but it did not work. What else needs to be saved and/or updated in order for this work? Thanks!

I'm using MuJoCo version 2.3.1 and Python bindings.

**Comment by yuvaltassa** [comment]
Go though our [tutorial colab](https://github.com/deepmind/mujoco#getting-started). If you still have this question, write back here.
